ext/mysqli/mysqli.stub.php

```diff
@@ -52,7 +52,7 @@ public function close() {}
      * @return bool
      * @alias mysqli_commit
      */
-    public function commit(int $flags = -1, string $name = UNKNOWN) {}
+    public function commit(int $flags = -1, string $name = "") {}
```

Comment:
There is an odd discrepancy with begin_transaction here (though I have no idea if the "name" they reference is the same).

Generally I'm finding it hard to decide what should be nullable and what should use empty strings...

Reply:
> Generally I'm finding it hard to decide what should be nullable and what should use empty strings...

I had the same problem. :/

> Though I have no idea if the "name" they reference is the same

While `begin_transaction()`'s `$name` is the `savepoint name for the transaction`, on the other hand, `commit()`'s has the following description: `If provided then COMMIT/*name*/ is executed`. So they doesn't seem the same for me.

I *feel* that a `""` default would be ok for both parameters, but in this case the `Savepoint name cannot be empty` warning should be removed (as written in https://github.com/php/php-src/pull/5514/files#r419069571). I'm also ok with making the two parameters nullable both, but probably I like this a bit less.

What do you prefer?

Reply:
I think I'd go with nullable parameters, also as a general rule. Unless empty string is somehow "naturally" the default (e.g. $suffix=""), using null is a more explicit encoding that the parameter has a special value.

Reply:
Ok. And I also fixed the `$name` optional param of `rollback()`.